### PR TITLE
Added a 'type' parameter to createInput()

### PR DESCRIPTION
**Pull request by wmmnola**
this should allow for the creation of different html input types(ie passwords).

### CODE REVIEW

File: lib/addons/p5.dom.js [hunk 1]
```diff
@@ -736,9 +737,11 @@
    *
    * </code></div>
    */
-  p5.prototype.createInput = function(value) {
+  p5.prototype.createInput = function(value, type) {
     var elt = document.createElement('input');
-    elt.type = 'text';
+    if (type) elt.type = type;
```
Comment (lmccart) [inline lib/addons/p5.dom.js]: let's condense this a little more:
`elt.type = type ? type : 'text';`

Comment (wmmnola) [inline lib/addons/p5.dom.js]: I didn't even know you could do that. Forgive me, I'm fairly new to Javascript.

Comment (lmccart) [inline lib/addons/p5.dom.js]: hehe don't worry about it! kind of neat though, right?

Comment (wmmnola) [inline lib/addons/p5.dom.js]: Yeah, it reminds me of the operators for the UNIX shell.

Comment (wmmnola) [inline lib/addons/p5.dom.js]: Sorry, Didnt realize I could edit the pull request. I implemented your code change. This is all very new to me.

**Comment by lmccart** [comment]
thanks!

**Comment by anurupmitra** [comment]
I assume
`createInput("Password", password)`
should work - but it doesn't. Am I doing anything incorrectly?

**Comment by wmmnola** [comment]
I believe it should be

createInput("Password","password")

**Comment by anurupmitra** [comment]
Thank you.

I deleted the question here as I thought that this was a dedicated pull request thread? Sorry, I am not a programmer and am new to github, so being a bit clumsy.

Have moved it here:
[](https://github.com/processing/p5.js/issues/1830)

**Comment by wmmnola** [comment]
Thats fine. You should be alright to comment on the Pull Request.

**Pull request merged**